apps/oxlint/src/lint.rs

```diff
@@ -561,6 +526,26 @@ impl LintRunner {
     }
 }
 
+fn print_and_flush_stdout(stdout: &mut dyn Write, message: &str) {
+    stdout.write_all(message.as_bytes()).or_else(check_for_writer_error).unwrap();
+    stdout.flush().unwrap();
+}
+
+fn check_for_writer_error(error: std::io::Error) -> Result<(), std::io::Error> {
+    // Do not panic when the process is killed (e.g. piping into `less`).
+    if matches!(error.kind(), ErrorKind::Interrupted | ErrorKind::BrokenPipe) {
+        Ok(())
+    } else {
+        Err(error)
+    }
+}
+
+fn render_report(handler: &GraphicalReportHandler, diagnostic: &OxcDiagnostic) -> String {
```

Comment:
Thanks for simplifying it. This was always something that bothered me. ❤️ 

This is a bit deeper offtopic: In my perfect world without problems:
`GraphicalReportHandler` should be used only in `output_formatter::default`. 
The JSON output wants report an error in JSON. 
We should still provide a default implementation for other output formats.

Reply:
yeah, I agree, the concerns here are a little bit mixed. I thought about trying to refactor this with the linter runtime work as well, but decided to leave it.